test/test_dynproc.py

```diff
@@ -164,14 +164,29 @@ def testJoin(self):
         size = MPI.COMM_WORLD.Get_size()
         rank = MPI.COMM_WORLD.Get_rank()
         server = client = address = None
-        # crate server/client sockets
+        host = socket.gethostname()
+        addresses = socket.getaddrinfo(host, None)
+        address_families = [ a[0] for a in addresses if a[1]==socket.SOCK_STREAM ]
+        # if both INET and INET6 are available, don't assume the order
+        # is the same on both server and client. Select INET if available.
+        if socket.AF_INET in address_families:
+            socket_family = socket.AF_INET
+        elif socket.AF_INET6 in address_families:
+            socket_family = socket.AF_INET6
+        elif socket.AF_UNIX in address_families:
+            socket_family = socket.AF_UNIX
+        elif address_families:
```

Comment:
I would remove this branch handling undocumented families.

Reply:
Can be removed, or the AF_UNIX case could be removed, leaving this else clause to catch "other" families.
The motivation for it was in case some funky new internet family gets invented in the future, AF_QUANTUM say.